nmdc_runtime/api/endpoints/allowances.py

```diff
@@ -0,0 +1,144 @@
+    """
+    if not check_action_permitted(user.username, "/allowances"):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Only admin users are allowed to issue /allowances commands.",
+        )
+
+
+@router.get("/allowances")
+def list_allowances(
+    username: Optional[str] = Query(None, description="Filter allowances by username"),
+    action: Optional[AllowanceActions] = Query(
+        None, description="Filter allowances by action"
+    ),
+    user: User = Depends(get_current_active_user),
+    mdb: Database = Depends(get_mongo_db),
+):
+    """
+    Get list of allowances, optionally filtered by username and/or action.
+
+    - If both username and action are provided, returns the specific allowance (if any).
+    - If only username is provided, returns all allowances for that user.
+    - If only action is provided, returns all allowances for that action.
+    - If neither is provided, returns all allowances.
+
+    """
+    check_can_run_allowances_endpoints(user)
+    filter_criteria = {}
+    if username:
+        filter_criteria["username"] = username
+    if action:
+        filter_criteria["action"] = action.value
+    allowances = list(
+        mdb["_runtime.api.allow"].find(
+            filter=filter_criteria,
+        )
+    )
+    rv = {}
+    rv["resources"] = [strip_oid(d) for d in allowances]
+    return rv
+
+
+@router.post(
+    "/allowances", response_model=Allowance, status_code=status.HTTP_201_CREATED
+)
+def create_allowance(
+    allowance: Allowance,
+    user: User = Depends(get_current_active_user),
+    mdb: Database = Depends(get_mongo_db),
+):
+    """
+    Create an allowance for a specific user and action.
+
+    **Note:** This endpoint is only accessible to authenticated users.
+    """
+    check_can_run_allowances_endpoints(user)
+    # Check if the allowance already exists
+    existing = mdb["_runtime.api.allow"].find_one(
+        {"username": allowance.username, "action": allowance.action}
+    )
+
+    if existing:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail=f"Allowance already exists for user '{allowance.username}' and action '{allowance.action}'",
+        )
+
+    # Create the allowance
+    doc = {"username": allowance.username, "action": allowance.action}
+    try:
+        mdb["_runtime.api.allow"].insert_one(doc)
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to create allowance: {str(e)}",
+        )
+
+    return Allowance(**doc)
+
+
+@router.delete("/allowances", status_code=status.HTTP_204_NO_CONTENT)
+def delete_allowance(
+    username: str = Query(..., description="Username for the allowance to delete"),
+    action: AllowanceActions = Query(
+        ..., description="Action for the allowance to delete"
+    ),
```

Comment:
Changed "for" to "of".

```suggestion
    username: str = Query(..., description="Username of the allowance to delete"),
    action: AllowanceActions = Query(
        ..., description="Action of the allowance to delete"
    ),
```